### PR TITLE
show uid_map in podman info

### LINKED ISSUE #4067: Rootless `podman Info` should show the current UID_MAP
Lots of times when people have problems with rootless podman, we find that they have not setup User Namespace correctly.

Having podman info display this data would allow us to eliminate one common request we give to users.

### PR DESCRIPTION
**Pull request by QiWang19**
show uid_map in podman info
close #4067 
Signed-off-by: Qi Wang <qiwan@redhat.com>

### CODE REVIEW

File: libpod/info.go [hunk 1]
```diff
@@ -128,6 +128,12 @@ func (r *Runtime) hostInfo() (map[string]interface{}, error) {
 	}
 	info["hostname"] = host
 	info["eventlogger"] = r.eventer.String()
+
+	uidmappings, err := rootless.ReadMappingsProc("/proc/self/uid_map")
+	if err != nil {
+		return nil, errors.Wrapf(err, "error reading uid mappings")
+	}
+	info["uidmap"] = uidmappings
```
Comment (mheon) [inline libpod/info.go]: This should only be done if we are rootless

Comment (giuseppe) [inline libpod/info.go]: I think we should also include "/proc/self/gid_map"

Comment (rhatdan) [reply libpod/info.go]: I agree

**Comment by QiWang19** [comment]
https://github.com/containers/libpod/issues/4067#issuecomment-533550391
I also added the log for `newuidmap/newgidmap`, when will the changes happen to newuidmap/newgidmap?
@mheon @giuseppe @rhatdan  PTAL

File: pkg/rootless/rootless_linux.go [hunk 2]
```diff
@@ -431,12 +431,14 @@ func becomeRootInUserNS(pausePid, fileToRead string, fileOutput *os.File) (bool,
...
 			return false, -1, errors.Wrapf(err, "cannot write setgroups file")
 		}
+		logrus.Debugf("write setgroups file exited with 0")
 
 		uidMap := fmt.Sprintf("/proc/%d/uid_map", pid)
 		err = ioutil.WriteFile(uidMap, []byte(fmt.Sprintf("%d %d 1\n", 0, os.Geteuid())), 0666)
 		if err != nil {
 			return false, -1, errors.Wrapf(err, "cannot write uid_map")
 		}
+		logrus.Debugf("write write uid_map exited with 0")
```
Comment (rhatdan) [inline pkg/rootless/rootless_linux.go]: write write?  Do we really need this debug information?

**Comment by rhatdan** [comment]
How does the output look when you run podman info?  Are the line separations ok?

**Comment by QiWang19** [comment]
> How does the output look when you run podman info? Are the line separations ok?

It directly prints out the IDMap struct
```
$ podman info
host:
  ... 
  IDMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  MemFree: 1369817088
  ...
```

File: pkg/rootless/rootless_linux.go (hunk 2 above)
Comment (QiWang19) [reply pkg/rootless/rootless_linux.go]: required by the comment under issue https://github.com/containers/libpod/issues/4067#issuecomment-533550391

**Comment by TomSweeneyRedHat** [comment]
The output looks good, but I'm concerned that if there are several hundred or more mappings on a system, then `podman info` will have an enormous amount of output.  Could we create a '--quiet'  option for `podman info` that would not show all of this, or at that time perhaps just show a count for the number of uid-map entries found?  Maybe instead have a `--verbose` option that would only show this info at that time?  @mheon WDYT?

**Comment by TomSweeneyRedHat** [comment]
Code LGTM other than my voluminous output concern.

**Comment by rhatdan** [comment]
@TomSweeneyRedHat I think you are creating a solution to a problem that does not exist.  I believe you used to be able to specify only 5 ranges, maybe you can specify more now.  But bottom line, unless someone is trying to cause problems, we should never see more then 5 lines
/lgtm
/approve

**Pull request merged**